### PR TITLE
[PW-7958] Generic Type in Service to reduce boiler plate code

### PR DESCRIPTION
**Pull request by jillingk**
**Description**
Adds support for generic response type in the ServiceResource class to reduce boilerplate code.
**Tested scenarios**
UT and IT

### CODE REVIEW

File: Adyen/Service/Checkout.cs [hunk 1]
```diff
@@ -293,8 +292,7 @@ public CreateCheckoutSessionResponse Sessions(CreateCheckoutSessionRequest creat
         public async Task<CreateCheckoutSessionResponse> SessionsAsync(CreateCheckoutSessionRequest createCheckoutSessionRequest)
         {
             var jsonRequest = createCheckoutSessionRequest.ToJson();
-            var jsonResponse = await _sessions.RequestAsync(jsonRequest);
```
Comment (wboereboom) [inline Adyen/Service/Checkout.cs:296]: Why not update the entire class? That way we could also maybe remove the dependency for this from this class

Comment (AlexandrosMor) [reply Adyen/Service/Checkout.cs:296]: Indeed it is only in one place :(

Comment (jillingk) [reply Adyen/Service/Checkout.cs:296]: Ah yeah I just wanted to test it and thought we'd change the rest later during the service automation. I can change it manually as well though!

Comment (jillingk) [reply Adyen/Service/Checkout.cs:296]: Changed it :)

**Review by Kwok-he-Chu — COMMENTED** [review]
I've given my input on the PR. Good work on it so far. I think we can do even better. One thing that came up to my mind was: Do we need a separate `ServiceResource`? The logic seems to be part of a client implementation that does the actual request. Something like this:

```csharp
public interface ISomeClient.cs // The implemementation should have the HttpClient that performs the actual request.
{
     Task<T> RequestAsync(string json, RequestOptions requestOptions = null, HttpMethod httpMethod = null, CancellationToken cancellationToken = default);
     T Request(string json, RequestOptions requestOptions = null, HttpMethod httpMethod = null, CancellationToken cancellationToken = default);
}
```


**Edit:** I just noticed that ServiceResource acts like an abstract class for all the services (for example [this](https://docs.google.com/document/d/1wGmZRS1SXGQgLsDkmQl_OEJYHhqCNS99stdxFZWoMco/edit). We should consider rename the class (by adding a suffix) to reflect its behavior. If you decide on refactoring this part, we should pick it up in a different ticket. I think all is good for now 👍

File: Adyen/Service/ServiceResource.cs [hunk 2]
```diff
@@ -4,6 +4,7 @@
 using System.Net.Security;
 using System.Threading.Tasks;
 using Adyen.Model;
+using Newtonsoft.Json;
 
 namespace Adyen.Service
```
[the remark this answers is not in the document]
Comment (Kwok-he-Chu) [reply Adyen/Service/ServiceResource.cs:9]: I noticed that we're introducing newtonsoft.json here, can we consider using System.Text.Json?

_Note: I'm aware that the library already uses Newtonsoft.jJon, if that's the only thing that you're using right now, I would keep it consistent, keep using Newtonsoft.Json and refactor this when the team is ready to move to Systen.Text.Json._

File: Adyen/Service/ServiceResource.cs [hunk 3]
```diff
@@ -23,11 +24,25 @@ public string Request(string json, RequestOptions requestOptions = null, HttpMet
...
+            return JsonConvert.DeserializeObject<T>(jsonResponse);
+        }
 
         public Task<string> RequestAsync(string json, RequestOptions requestOptions = null, HttpMethod httpMethod = null)
         {
             var clientInterface = _abstractService.Client.HttpClient;
             return clientInterface.RequestAsync(Endpoint, json, requestOptions, httpMethod);
         }
+        
+        public async Task<T> RequestAsync<T>(string json, RequestOptions requestOptions = null, HttpMethod httpMethod = null)
```
Comment (Kwok-he-Chu) [inline Adyen/Service/ServiceResource.cs:41]: Pass on a CancellationToken here for asynchronous method.

```csharp
 public async Task<T> RequestAsync<T>(string json, RequestOptions requestOptions = null, HttpMethod httpMethod = null, CancellationToken cancellationToken = default)
```

**Review by michaelpaul — APPROVED** [review]

**Review by AlexandrosMor — APPROVED** [review]

**Pull request merged**